test/e2e/puppeteer.js

```diff
@@ -246,6 +241,7 @@ async function main() {
 	/* Launch browser */
 
 	const flags = [ '--hide-scrollbars', '--enable-gpu' ];
+	flags.push( '--enable-unsafe-webgpu', '--enable-webgpu-developer-features', '--use-angle=swiftshader', '--user-agent=puppeteer' );
```

Comment:
try `--use-angle=metal`. Also, you probably don't need `--user-agent=puppeteer`. I had that in some rule so that some other code could tell if it was running in puppeteer or another browser.

Reply:
Also, just FYI, I added some code to both twgl's tests and webgpu-utils's tests to log the browser info (from navigator.userAgentData) and the GPU info (from WEBGL_debug_renderer_info in WebGL and from adapter.info, adapter.requestAdapterInfo in WebGPU. It let me see that it was using WebGL with OpenGL instead of Metal on Windows. It also let me see what version of chrome puppeteer is using. You might want to do the same.